test/powershell/Language/Scripting/TryCatch.Tests.ps1

```diff
@@ -0,0 +1,651 @@
+            }
+        }
+    }
+    
+    It "Test simple parsing, ensure newlines allowed everywhere" {
+        {
+            try
+            {
+            }
+            catch
+            {
+            }
+
+            try
+            {
+            }
+            catch
+            [int]
+            {
+            }
+
+            try
+            {
+            }
+            catch
+            [int]
+            ,
+            [char]
+            {
+            }
+
+            try
+            {
+            }
+            finally
+            {
+            }
+
+            try
+            {
+            }
+            catch
+            {
+            }
+            finally
+            {
+            }
+
+            try
+            {
+            }
+            catch
+            [int]
+            {
+            }
+            finally
+            {
+            }
+
+            try
+            {
+            }
+            catch
+            [int]
+            ,
+            [char]
+            {
+            }
+            finally
+            {
+            }
+        } | Should Not Throw
+    }
+
+    Context "Basic exception handling" {
+        It "Simple throw and catch" {
+            $a = . { try { 1; throw "exception"; "test failed" } catch { 2 } }
+            AssertArraysEqual $a (1, 2)
+        }
+
+        It "Simple try finally" {
+            $a = . { try { 1 } finally { 2 } }
+            AssertArraysEqual $a (1,2)
+        }
+
+        It "Simple try, throw, catch, and finally" {
+            $a = . { try { 1; throw "exception"; "test failed" } catch { 2 } finally { 3 } }
+            AssertArraysEqual $a (1..3)
+        }
+    }
+
+    Context "Mix traps with try/catch" {
+        It "Trap shouldn't catch exception" {
+            $a = . { trap { "test failed" } try { 1; throw "exception"; "test failed" } catch { 2 } }
+            AssertArraysEqual $a (1,2)
+        }
+
+        It "Trap should catch exception" {
+            $a = . { try { 1; throw "exception"; trap { 2; return }; "test failed" } catch { "test failed" } }
+            AssertArraysEqual $a (1,2)
+        }
+    }
+
+    Context "Catch by type" {
+        It "Catch by type #1" {
+            $a = . { try { 1; $a = 0; 1/$a; "test failed" } catch [DivideByZeroException] { 2 } }
+            AssertArraysEqual $a (1,2)
+        }
+
+        It "Catch by type #2" {
+            $a = . { try { 1; $a = 0; 1/$a; "test failed" } catch [DivideByZeroException] { 2 } catch [Exception] { "test failed" } }
+            AssertArraysEqual $a (1,2)
+        }
+
+        It "Catch by type #3" {
+            $a = . { try { 1; $a = 0; 1/$a; "test failed" } catch [DivideByZeroException] { 2 } catch { "test failed" } }
+            AssertArraysEqual $a (1,2)
+        }
+
+        It "Catch by type #4" {
+            $a = . { try { 1; $a = 0; 1/$a; "test failed" } catch [DivideByZeroException],[ArgumentNullException] { 2 } }
```

Comment:
yah - since the test immediately prior is catching `DivideByZeroException` I wasn't sure what this test was really doing. Should we have two tests? One to catch `DivideByZeroException` and on to catch `ArgumentNullException` (e.g., we can catch both exception types) - if we can improve on our existing tests and make them better/clearer that's awesome